### ISSUE TITLE
Weird things appear in autocomplete

### ISSUE THREAD

**Issue by MewPurPur** [godotengine/godot#76919]
### Godot version

4.0.2

### System information

Ubuntu 22.04.1 LTS

### Issue description

So far I've found these ones:

![Screenshot from 2023-05-10 17-56-42](https://github.com/godotengine/godot/assets/85438892/b161b14a-cb7d-46fe-8398-816bf354a929)
![Screenshot from 2023-05-10 18-01-02](https://github.com/godotengine/godot/assets/85438892/b0fefcfb-4246-4ebb-ae7b-dd798d7a1d63)

But there are likely more like them. They are invalid, at least in the context I encountered them here (at the beginning of a regular line of script)

### Steps to reproduce

Type `Anchor` in the code editor.

### Minimal reproduction project

N/A

**Comment by Sauermann** [comment]
Looks like a separated substring search:
- `ANC`
  - `get_pAreNt_Control`
- `CONTAIN`
  - `is_loCal_transfOrm_NoTificAtIoN_enabled`
  - `update_CONfiguraTion_wArnINgs`

**Comment by kleonc** [comment]
The behavior might be indeed intended but highlighting the matching part is clearly incorrect in some cases.

**Comment by MewPurPur** [comment]
Yes, the highlighting is awkward, but look at "Anchor Offsets". It's not a property

**Comment by ajreckof** [comment]
Will be fixed when #75746 is merged?

**Comment by lmsonic** [comment]
Tested on 4.1.1, can reproduce the Container Sizing, Anchor Offset and Anchor Points non existing properties showing up, whilst the highlighting and search looks fixed.
![image](https://github.com/godotengine/godot/assets/25965198/fee4fd69-8be3-4688-8776-ac3186dd4157)
![image](https://github.com/godotengine/godot/assets/25965198/899bfccf-e9ba-4e96-a0b8-949365f2ea32)

**Comment by AThousandShips** [comment]
Those are categories, should be filtered out
